hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/OzoneTokenIdentifier.java

```diff
@@ -77,6 +80,122 @@ public Text getKind() {
     return KIND_NAME;
   }
 
+  /** Instead of relying on proto serialization, this
+   *  provides  explicit serialization for OzoneTokenIdentifier.
+   * @return byte[]
+   */
+  public byte[] toUniqueSerializedKey() {
+    ByteBuffer result =
+        ByteBuffer.allocate(4096);
```

Comment:
I see there is StrToSign if it's S3 auth. I 'm curious about how big this StrToSign can be? And whether 4096 bytes can hold all these.

Reply:
> I see there is StrToSign if it's S3 auth. I 'm curious about how big this StrToSign can be? And whether 4096 bytes can hold all these.

It should not be too big and not cause overflow.

https://docs.aws.amazon.com/general/latest/gr/sigv4-create-string-to-sign.html

I saw something similar in other places too e.g. 
org/apache/hadoop/security/token/TokenIdentifier.class with 4096 bytes limit.

Reply:
@ChenSammi Are you OK with resolving this issue?

Reply:
This issue is no longer valid in the newer version of the patch. cc @ChenSammi.